test/geometry/transform/test_image_registrator.py

```diff
@@ -0,0 +1,89 @@
+import pytest
+import torch
+from torch.autograd import gradcheck
+
+import kornia
+import kornia.testing as utils  # test utils
+from kornia.geometry.transform import ImageRegistrator
+from kornia.testing import assert_close
+
+
+class TestSimilarity:
+    def test_smoke(self, device, dtype):
+        expected = torch.eye(3)[None]
+        for r, sc, sh in zip([True, False], [True, False], [True, False]):
+            sim = kornia.geometry.transform.Similarity(r, sc, sh)
+            assert_close(sim(), expected, atol=1e-4, rtol=1e-4)
+
+    def test_smoke_inverse(self, device, dtype):
+        expected = torch.eye(3)[None]
+        for r, sc, sh in zip([True, False], [True, False], [True, False]):
+            s = kornia.geometry.transform.Similarity(r, sc, sh)
+            assert_close(s.forward_inverse(), expected, atol=1e-4, rtol=1e-4)
+
+    def test_scale(self, device, dtype):
+        sc = 0.5
+        sim = kornia.geometry.transform.Similarity(True, True, True)
+        sim.scale.data *= sc
+        expected = torch.tensor([[0.5, 0, 0.],
+                                [0, 0.5, 0],
+                                [0, 0, 1]])[None]
+        inv_expected = torch.tensor([[2.0, 0, 0.],
+                                    [0, 2.0, 0],
+                                    [0, 0, 1]])[None]
+        assert_close(sim.forward_inverse(), inv_expected, atol=1e-4, rtol=1e-4)
+        assert_close(sim(), expected, atol=1e-4, rtol=1e-4)
+
+    def test_repr(self, device, dtype):
+        for r, sc, sh in zip([True, False], [True, False], [True, False]):
+            s = kornia.geometry.transform.Similarity(r, sc, sh)
+            print(s)
+
+
+class TestHomography:
+    def test_smoke(self, device, dtype):
+        expected = torch.eye(3)[None]
+        h = kornia.geometry.transform.Homography().to(device)
+        assert_close(h(), expected, atol=1e-4, rtol=1e-4)
+
+    def test_smoke_inverse(self, device, dtype):
+        expected = torch.eye(3)[None]
+        h = kornia.geometry.transform.Homography().to(device)
+        assert_close(h.forward_inverse(), expected, atol=1e-4, rtol=1e-4)
+
+    def test_repr(self, device, dtype):
+        h = kornia.geometry.transform.Homography().to(device)
+        print(h)
+
+
+class TestImageRegistrator:
+    def test_smoke(self, device, dtype):
+        for model_type in ['homography',
+                           'similarity',
+                           'translation',
+                           'scale',
+                           'rotation']:
+            ir = kornia.geometry.transform.ImageRegistrator(model_type)
+            print(ir)
+
+    def test_registration(self, device):
```

Comment:
```suggestion
    def test_registration(self, device, dtype):
```
missing or any other reason ?

Reply:
you cannot optimize over non-float variable. I first added it as you suggested, but tests were failing for this reason.